msteams-platform/apps-in-teams-meetings/meeting-apps-apis.md

```diff
@@ -282,6 +284,96 @@ The following table includes the response codes:
 | **403** | The app is unable to send the signal. 403 response code can occur because of various reasons, such as the tenant admin disables and blocks the app during live site migration. In this case, the payload contains a detailed error message. |
 | **404** | The meeting chat doesn't exist. |
 
+## Targeted in-meeting notification API
+
+The `targetedMeetingNotification` API allows apps to send notifications to specific participants in a meeting. Apps send targeted in-meeting notifications based on user action. The API is available through bot API.
+
+> [!NOTE]
+>
+> * The API payload only permits a task module with a URL.
+> * The user ID formats **aadObjectid** and **UPN** aren't supported.
+
+Get supported user ID format for targeted in-meeting notification:
+
+* [Get participant API](#get-participant-api)
+* [Get members API](../bots/how-to/get-teams-context.md#fetch-the-roster-or-user-profile)
+
+### Example
+
+Following is an example of request payload:
+
+```http
+POST /v1/meetings/{meetingId}/notification
+```
+
+```json
+{
+
+  "type": "targetedMeetingNotification",
+  "value": {
+    "recipients": [ 
+"29:1I12M_iy2wTa97T6LbjTh4rJCWrtw2PZ3lxpD3yFv8j2YPnweY2lpCPPAn3RI0PP7rghfHauUz48I1t7ANhj4CA"
+     ], 
+    "surfaces": [ 
+      { 
+        "surface": "meetingStage", 
+        "contentType": "task", 
+        "content": { 
+          "value": { 
+            "height": "300", 
+            "width": "400", 
+            "title": "Targeted meeting Notification", 
+            "url": "https://somevalidurl.com"           
+}
+        } 
+      } 
+    ] 
+  },
+  "channelData": { // optional if a developer doesn't want to support user attributes.
+    "onBehalfOf": [ 
+      { 
+        "itemid": 0, 
+        "mentionType": "person", 
+        "mri": "29:1mDOCfGM9825lMHlwP8NjIVMJeQAbN-ojYBT5VzQfPpnst1IFQeYB1QXC8Zupn2RhgfLIW27HmynQk-4bdx_YhA", 
+        "displayName": "yunny chung"      } 
+    ] 
+  }
+}
+```
+
+| Property name | Description |
+|---|---|
+| `meetingId` | The meeting ID is available through bot invoke and TeamsJS library. |
+| `type` |`targetedMeetingNotification` |
+| `recipients` | List of user IDs. Get user IDs for meeting participants through [Get participant API](#get-participant-api). Get the entire list of chat roster using [Get members API](../bots/how-to/get-teams-context.md#fetch-the-roster-or-user-profile). Empty or null recipients list will return 400.|
+| `surface` | `meetingStage` keyword. |
```

Comment:
Thanks for adding this table! 
I initially wrote this doc but I got feedback from my teammate that this description is confusing as it does not capture the extensibility capability of the API. 

Can we update the table like below? 

- `Surfaces`: List of surfaces where notifications can be rendered.
- `Surface`: A type of surface. Currently the only supported surface type is `meetingStage`. 
-  `ContentType`: type of content that the targeted in-meeting notification will render. Currently supported value is `task`.
- `Content`: [TaskModuleContinueResponse](if there is a doc link about this object, let's link it.)

Reply:
updated, task module need to check the link.